### PR TITLE
Remove japanese locale

### PR DESCRIPTION
**Pull request by CarlosCortizasCT**
## Summary

Remove Japanese locale as we're not supporting it anymore.

## Description

Since this locale is not being used, we would like to avoid confusion and reduce costs.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,7 +17,7 @@
     "build": "./scripts/build.sh",
     "build:watch": "./scripts/build_watch.sh",
     "start": "yarn --cwd docs start",
-    "extract-intl": "formatjs extract --format=./packages/i18n/transifex-transformer.js --out-file=./packages/i18n/data/core.json '{packages,src/components}/**/messages.js'",
+    "extract-intl": "formatjs extract --format=./packages/i18n/transifex-transformer.js --out-file=./packages/i18n/data/core.json '{packages,src/components}/**/messages.ts'",
```
Comment (CarlosCortizasCT) [inline package.json:20]: This change is not related, but I noticed the command was wiping out all translations when I tried to use it to check the other changes.

**Review by emmenko — APPROVED** [review]
Thanks 🥳

**Comment by emmenko** [comment]
PS: the changeset is missing

**Review by emmenko — APPROVED** [review]

**Review by ddouglasz — APPROVED** [review]
Looks good!

**Pull request merged**